core/src/cluster_info.rs

```diff
@@ -1729,37 +1731,51 @@ impl ClusterInfo {
         Ok(())
     }
 
-    fn handle_adopt_shred_version(self: &Arc<Self>, adopt_shred_version: &mut bool) {
-        // Adopt the entrypoint's `shred_version` if ours is unset
-        if *adopt_shred_version {
-            // If gossip was given an entrypoint, look up the ContactInfo by the given
-            // entrypoint gossip adddress
-            let gossip_addr = self.entrypoint.read().unwrap().as_ref().map(|e| e.gossip);
+    fn process_entrypoints(self: &Arc<Self>, entrypoints_processed: &mut bool) {
```

Comment:
Changes here:
* adopt shred version from the first non-zero entrypoint shred version
* update the entrypoint.id for all entrypoints.  Don't set `entrypoints_processed = true` until they've all been updated